### ISSUE TITLE
Invalid signature

### ISSUE THREAD

**Issue by cloudhead** [radicle-dev/radicle-upstream#1138]
### Missing refs
```
/home/cloudhead/.local/share/radicle/git/refs/namespaces/hwd1yreg6n7jb8mtaia9tpww8q5jyr38rxydeodbyyebkkeu45b974wc1bw
└── refs
    ├── heads
    │   └── main
    ├── rad
    │   ├── id
    │   ├── ids
    │   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
    │   ├── self
    │   └── signed_refs
    └── remotes
        ├── hydr574fy5ujmjetw95e6jqexs94o179qs8wpdrsnzqfmuo7jpwbbg
        │   └── rad
        │       └── signed_refs
        ├── hyyactdhac4ijg7613nqehj5qw3dwq4bjpett45z9aqjzgsr7rkf4s
        │   └── rad
        │       └── signed_refs
        └── hyywqooy513gro7homjfzix4nwopcipgkjzc653os5w3sps3o8k73c
            └── rad
                └── signed_refs
```

**Comment by cloudhead** [comment]
[buggy-radicle.tar.gz](https://github.com/radicle-dev/radicle-upstream/files/5464462/buggy-radicle.tar.gz)

**Comment by cloudhead** [comment]
[buggy-librad.key.tar.gz](https://github.com/radicle-dev/radicle-upstream/files/5464488/buggy-librad.key.tar.gz)

**Comment by rudolfs** [comment]
As seen from my point of view:

[console.log](https://github.com/radicle-dev/radicle-upstream/files/5464546/console.log)
[xyz.radicle.radicle-upstream.tar.gz](https://github.com/radicle-dev/radicle-upstream/files/5464547/xyz.radicle.radicle-upstream.tar.gz)
[xyz.radicle.radicle.tar.gz](https://github.com/radicle-dev/radicle-upstream/files/5464548/xyz.radicle.radicle.tar.gz)


### observations
  - no way to find out our own PeerId from the upstream UI
  - copying the urn from the seed `rad:git:hwd1yreg6n7jb8mtaia9tpww8q5jyr38rxydeodbyyebkkeu45b974wc1bw/main` can't be pasted in upstream search because it contains `/main`
  - would be nice to have a back button to get back to the project list


### bugs
  - you're connected to 5 peers although I'm only connected to the seed node
  - going from another project to a cloned project gives us an "Incorrect input"
  - peer selector doesn't have all peers, but they're present in the manage remotes screen
    -> peer selector only shows myself if i pushed any changes to a foreign repo
    -> we probably shouldn't show ourselves in the remote modal
  - choosing a file in a different repo then switching to another one
```
FileSource.svelte:100 Uncaught (in promise) TypeError: Cannot read property 'split' of null
    at create_fragment$2l (FileSource.svelte:100)
    at init (index.mjs:1447)
    at new FileSource (FileSource.svelte:98)
    at Array.create_if_block$B (Source.svelte:125)
    at Array.create_default_slot$1w (Source.svelte:111)
    at create_slot (index.mjs:58)
    at Array.create_else_block$4 (Remote.svelte:34)
    at Array.create_if_block_2$5 (Remote.svelte:34)
    at create_fragment$1n (Remote.svelte:39)
    at init (index.mjs:1447)
```

**Comment by juliendonck** [comment]
linked #1145

**Comment by juliendonck** [comment]
#1144

**Comment by xla** [comment]
State on the seed for the project:

```
/state/git/refs/namespaces/hwd1yreg6n7jb8mtaia9tpww8q5jyr38rxydeodbyyebkkeu45b974wc1bw
└── refs
    ├── rad
    │   ├── id
    │   ├── ids
    │   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
    │   └── signed_refs
    └── remotes
        ├── hydr574fy5ujmjetw95e6jqexs94o179qs8wpdrsnzqfmuo7jpwbbg
        │   ├── heads
        │   │   └── main
        │   └── rad
        │       ├── id
        │       ├── ids
        │       │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
        │       ├── self
        │       └── signed_refs
        ├── hyyactdhac4ijg7613nqehj5qw3dwq4bjpett45z9aqjzgsr7rkf4s
        │   └── rad
        │       └── signed_refs
        ├── hyymt983ehun66uh196598dq9r7qumnkw3eqpobkam4ebydwzm1oao
        │   ├── heads
        │   │   └── main
        │   └── rad
        │       ├── id
        │       ├── ids
        │       │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
        │       ├── self
        │       └── signed_refs
        └── hyywqooy513gro7homjfzix4nwopcipgkjzc653os5w3sps3o8k73c
            ├── heads
            │   ├── main
            │   └── merle
            │       └── lolz
            └── rad
                ├── id
                ├── ids
                │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
                ├── self
                └── signed_refs
```

**Comment by FintanH** [comment]
Rudolfs' view looks like:

```
refs/namespaces/
├── hwd1yre8988ax9ok6qn5xbhd98r1a7rtbe5uiic8xsomzxj5rsib4etxuyh
│   └── refs
│       ├── heads
│       │   └── master
│       └── rad
│           ├── id
│           ├── ids
│           │   └── hwd1yrebpnrqnuxhgkm6s4ngqa95rujzn64xrrnn9fub71hqi4b7spasoba
│           ├── self
│           └── signed_refs
├── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
│   └── refs
│       ├── rad
│       │   └── id
│       └── remotes
│           └── hyymt983ehun66uh196598dq9r7qumnkw3eqpobkam4ebydwzm1oao
│               └── rad
│                   └── id
├── hwd1yrebpnrqnuxhgkm6s4ngqa95rujzn64xrrnn9fub71hqi4b7spasoba
│   └── refs
│       └── rad
│           ├── id
│           ├── self
│           └── signed_refs
├── hwd1yrefgq7xcfhw5pxrq1d4huigij541owrwjphpgiqji1sipemwtoejor
│   └── refs
│       ├── rad
│       │   ├── id
│       │   ├── ids
│       │   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
│       │   ├── self
│       │   └── signed_refs
│       └── remotes
│           ├── hynewpywqj6x4mxgj7sojhue3erucyexiyhobxx4du9w66hxhbfqbw
│           │   └── rad
│           │       ├── id
│           │       └── ids
│           │           └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
│           └── hyymt983ehun66uh196598dq9r7qumnkw3eqpobkam4ebydwzm1oao
│               ├── heads
│               │   └── main
│               └── rad
│                   ├── id
│                   ├── ids
│                   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
│                   ├── self
│                   └── signed_refs
└── hwd1yreg6n7jb8mtaia9tpww8q5jyr38rxydeodbyyebkkeu45b974wc1bw
    └── refs
        ├── heads
        │   └── main
        ├── rad
        │   ├── id
        │   ├── ids
        │   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
        │   ├── self
        │   └── signed_refs
        └── remotes
            ├── hynewpywqj6x4mxgj7sojhue3erucyexiyhobxx4du9w66hxhbfqbw
            │   └── rad
            │       ├── id
            │       └── ids
            │           └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
            ├── hyyactdhac4ijg7613nqehj5qw3dwq4bjpett45z9aqjzgsr7rkf4s
            │   └── rad
            │       └── signed_refs
            ├── hyymt983ehun66uh196598dq9r7qumnkw3eqpobkam4ebydwzm1oao
            │   ├── heads
            │   │   └── main
            │   └── rad
            │       ├── id
            │       ├── ids
            │       │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
            │       ├── self
            │       └── signed_refs
            └── hyywqooy513gro7homjfzix4nwopcipgkjzc653os5w3sps3o8k73c
                └── rad
                    └── signed_refs
```

The project that is error'ing looks like:
```
refs/namespaces/hwd1yreg6n7jb8mtaia9tpww8q5jyr38rxydeodbyyebkkeu45b974wc1bw/
└── refs
    ├── heads
    │   └── main
    ├── rad
    │   ├── id
    │   ├── ids
    │   │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
    │   ├── self
    │   └── signed_refs
    └── remotes
        ├── hynewpywqj6x4mxgj7sojhue3erucyexiyhobxx4du9w66hxhbfqbw
        │   └── rad
        │       ├── id
        │       └── ids
        │           └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
        ├── hyyactdhac4ijg7613nqehj5qw3dwq4bjpett45z9aqjzgsr7rkf4s
        │   └── rad
        │       └── signed_refs
        ├── hyymt983ehun66uh196598dq9r7qumnkw3eqpobkam4ebydwzm1oao
        │   ├── heads
        │   │   └── main
        │   └── rad
        │       ├── id
        │       ├── ids
        │       │   └── hwd1yre8jcxu6hp9gjnr6z47zs5bx1g4qoowanatkhy9io6khbkbjgkcyrw
        │       ├── self
        │       └── signed_refs
        └── hyywqooy513gro7homjfzix4nwopcipgkjzc653os5w3sps3o8k73c
            └── rad
                └── signed_refs
```

One strange thing is that one remote doesn't even have a `signed_refs`. But that will just get ignored. Again, the error gets thrown by a `signed_refs` that exists just by itself.

**Comment by FintanH** [comment]
So this structure of only having `signed_refs` seems really off, e.g. below:
```
        └── hyywqooy513gro7homjfzix4nwopcipgkjzc653os5w3sps3o8k73c
            └── rad
                └── signed_refs
```

During a clone we first fetch `rad/id`, `rad/self`, and `rad/ids/*`. If these are "invalid" then we error out and we never fetch the `signed_refs`. So in what scenario would we have only `signed_refs`?

Well, I think since we recently started testing the tracking of other users that this might be the cause. If we track a remote we're adding them to the tracking graph and so we'll fetch their changes when they advertise them. Well, what happens if they do advertise commits? I'm betting that we end up with only their `signed_refs`.

I'm not convinced that this is the reason why we're getting "invalid signature" errors though. I wonder if somehow `signed_refs` by _another_ peer are getting replicated by the original peer? Something like `Peer A` has `signed_refs`. `Peer B` fetches these and then advertises them as their own. Finally, `Peer C` sees these `signed_refs` from `Peer B` but the signature doesn't match. I've no evidence for this, but it's a hunch I've got for now.
